dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs

```diff
@@ -69,22 +70,28 @@ public class OllamaAgent : IStreamingAgent
             while (!reader.EndOfStream && !cancellationToken.IsCancellationRequested)
             {
                 string? line = await reader.ReadLineAsync();
-                if (string.IsNullOrWhiteSpace(line)) continue;
+                if (string.IsNullOrWhiteSpace(line))
+                {
+                    continue;
+                }
 
                 ChatResponseUpdate? update = JsonSerializer.Deserialize<ChatResponseUpdate>(line);
-                if (update != null)
+                if (update is { Done: false })
                 {
                     yield return new MessageEnvelope<ChatResponseUpdate>(update, from: Name);
                 }
+                else
+                {
+                    var finalUpdate = JsonSerializer.Deserialize<ChatResponse>(line) ?? throw new Exception("Failed to deserialize response");
 
-                if (update is { Done: false }) continue;
-
-                ChatResponse? chatMessage = JsonSerializer.Deserialize<ChatResponse>(line);
-                if (chatMessage == null) continue;
-                yield return new MessageEnvelope<ChatResponse>(chatMessage, from: Name);
+                    yield return new MessageEnvelope<ChatResponse>(finalUpdate, from: Name);
+                }
             }
         }
     }
+
+    public string Name { get; }
```

Comment:
Nit : Can we move Name up on line 24-26 with the other fields?

Reply:
The name is a property